modules/platform/nuxeo-platform-document-routing/nuxeo-routing-core/src/test/java/org/nuxeo/ecm/platform/routing/test/TestDocumentRoutingService.java

```diff
@@ -361,6 +367,25 @@ public void testExecuteSimpleRouteWithConditionalStep() {
         assertEquals("canceled", children.get(2).getCurrentLifeCycleState());
     }
 
+    @Test
+    public void testOrphanTasksDeletion() {
+        DocumentModel route = session.createDocumentModel("/default-domain/workspaces", "dummyRoute",
+                DOCUMENT_ROUTE_DOCUMENT_TYPE);
+        route.setPropertyValue(TITLE_PROPERTY_NAME, "dummyRoute");
+        route = session.createDocument(route);
+
+        DocumentModel taskDocument = session.createDocumentModel("/default-domain/workspaces", "dummyTask",
+                TASK_TYPE_NAME);
+        taskDocument.setPropertyValue(TASK_PROCESS_ID_PROPERTY_NAME, route.getId());
+        taskDocument = session.createDocument(taskDocument);
+        session.save();
```

Comment:
not exactly sure but since I put postCommit to true and extend postCommitListener the test is broken, so I'm gonna look into it. Maybe I'll need `awaitility`